actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -344,6 +344,10 @@ def tag(name = nil, options = nil, open = false, escape = true)
       #   <% end -%>
       #    # => <div class="strong">Hello world!</div>
       def content_tag(name, content_or_options_with_block = nil, options = nil, escape = true, &block)
+        html_tag_regex = /^[a-zA-Z][\w-]*$/
```

Comment:
```suggestion
        html_tag_regex = /\A[a-zA-Z][\w-]*\z/
```

`^` and `$` match start and end of line.

`\A` and `\z` match start and end of string.

Also I don't think you need to assign it to a variable, just call match? directly on the literal.

Reply:
variable is added for the readability